### PR TITLE
External request handling

### PR DESCRIPTION
**Pull request by fatima3558**
## Overview

This PR updates code to handle requests from external sites more gracefully and sets timeout limits for these requests.

### Checklist

- [x] PR has a descriptive enough title to be useful in changelogs

## Testing Instructions

 * verify logic added makes sense
 * verify tests pass

Handles #699

### CODE REVIEW

**Comment by fatima3558** [comment]
the test that's failing is expecting a mocked response to have a `status_code` property. I'll update the test to give the mock request that property

File: docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: '2.4'
 services:
   app:
-    image: lametro:upgrade
+    image: ghcr.io/datamade/la-metro-councilmatic:master
```
Comment (hancush) [inline docker-compose.yml:4]: You love to see it!

File: lametro/models.py [hunk 2]
```diff
@@ -552,21 +552,26 @@ def _streaming_meeting(cls):
...
-            # comparison.
-            meeting = cls.objects.filter(extras__guid=guid.upper())
-
-            if meeting:
-                return meeting
+
+        try:
+            running_events = requests.get('http://metro.granicus.com/running_events.php', timeout=5)
+        except requests.exceptions.ConnectTimeout:
+            return cls.objects.none()
```
Comment (hancush) [inline lametro/models.py:559]: Yes, love the early return!

**Review by hancush — COMMENTED** [review]
Looking good. Can you propose a new test confirming that `current_meeting` returns the expected value when the live meeting endpoint resolves with a non-200 status code?

**Comment by fatima3558** [comment]
I can model it off of this test:

https://github.com/datamade/la-metro-councilmatic/blob/57b1dbc45e7b78f7aa272fd5b7d4bd022850beb9/tests/test_events.py#L124-L142

I would set the mock_request.status_code to be 404, keep the patch the same without the return_value (since this is what the assertion would test), remove the `current_meeting` references, and then assert that what is returned is an empty queryset.

**Comment by fatima3558** [comment]
@hancush I was referring to lines 136 and 139. The pseudocode would look something like this:

 def test_meeting_stream_link_unavailable(mocker): 
     mock_response = mocker.MagicMock(spec=requests.Response) 
     mock_response.status_code = 404
  
     mocker.patch('lametro.models.requests.get')
  
     assert empty queryset returned

**Comment by fatima3558** [comment]
as I'm working through this, I'm actually not sure what the assertion should be; I can't assert that I'm getting an empty queryset back because this would be part of the response object, which I'm currently mocking; based on what I should be asserting, the structure of the test will change. Do you have suggestions about what should I base my assertion/s off of, @hancush ?

**Comment by hancush** [comment]
@fatima3558 In general, that looks good. For your commented line, we do want to test that, if the streaming meetings request fails, the `current_meeting()` method returns an empty queryset. Does that make sense?

**Review by hancush — APPROVED** [review]
Nice job, @fatima3558, thank you!

**Pull request merged**